File: teams/suzumiya2014/week4_work.py
from flow_py_sdk import Script, flow_client
import asyncio
import json



async def get_nbatopshot_transaction_events():
    async with flow_client(
        host="access.mainnet.nodes.onflow.org", port=9000
    ) as client:
        block = await client.get_latest_block(is_sealed=True)
        startBlock = block.height-20
        endBlock = block.height
        event_type = "A.c1e4f4f4c4257510.Market.MomentPurchased"
        print(f"当前区块高度: {block.height}\n")
        print(f"下面查询高度{startBlock}至{endBlock}的MomentPurchased交易: \n")
        block_events = await client.get_events_for_height_range(
            type=event_type,
            start_height=startBlock,
            end_height=endBlock
        )
        for block_event in block_events:
            for event in block_event.events:
                fields = json.loads(event.payload)['value']['fields']
                id = fields[0]['value']['value']
                price = fields[1]['value']['value']
                seller = fields[2]['value']['value']['value']
                print(f"type: {event.type}")
                print(f"transaction_id: {event.transaction_id.hex()}")
                print(f"id: {int(id)}")
                print(f"price: {float(price)}")
                print(f"seller: {seller}")
                print('\n')

if __name__ == "__main__":
    asyncio.run(get_nbatopshot_transaction_events())

File: teams/SnC/week4_work.go

package main

import (
	"context"
	"fmt"
	"github.com/onflow/flow-go-sdk/client"
	"google.golang.org/grpc"
)

func main() {

	flowClient, err := client.New("access.mainnet.nodes.onflow.org:9000", grpc.WithInsecure())
	processErr(err)
	err = flowClient.Ping(context.Background())
	processErr(err)

	//latestBlock, err := flowClient.GetLatestBlock(context.Background(), false)
	//processErr(err)
	//fmt.Println("current block height: ", latestBlock.Height)

	blockEvents, err := flowClient.GetEventsForHeightRange(context.Background(), client.EventRangeQuery{
		Type:        "A.c1e4f4f4c4257510.Market.MomentListed",
		StartHeight: 14481080,
		EndHeight:   14481089,
	})
	processErr(err)

	for _, blockEvent := range blockEvents {
		for _, purchaseEvent := range blockEvent.Events {
			//fmt.Printf("%s\n\n",purchaseEvent.Value)
			fmt.Printf("Transaction ID: %s\n", purchaseEvent.TransactionID)
			fmt.Printf("Event ID: %s\n", purchaseEvent.ID())
			fmt.Printf("Moment ID: %d\n", purchaseEvent.Value.Fields[0])
			fmt.Printf("Price: %d\n", purchaseEvent.Value.Fields[1])
			fmt.Printf("Seller: %s\n", purchaseEvent.Value.Fields[2])

		}
	}
}

func processErr(err error) {
	if err != nil {
		fmt.Println("Err:" + err.Error())
		panic(err)
	}
}
